### PR TITLE
Update go and golangci versions

### PR DESCRIPTION
**Pull request by mgdelacroix**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/mattermost/mmctl
 
-go 1.13
+go 1.14
```
Comment (agnivade) [inline go.mod:3]: Wondering why is this needed? Unless we are raising the minimum required Go version to compile mmctl (which I don't think we are), there's no need to change this.

File: .circleci/config.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ executors:
   default:
     working_directory: ~/go/src/github.com/mattermost/mmctl
     docker:
-      - image: circleci/golang:1.13
+      - image: circleci/golang:1.14.6
```
Comment (agnivade) [inline .circleci/config.yml:10]: Let's go for 1.14.8

**Review by agnivade — COMMENTED** [review]
Just some small comments.

File: go.mod (hunk 1 above)
Comment (mgdelacroix) [reply go.mod]: Right! Changed back to `1.13`

**Review by isacikgoz — APPROVED** [review]
LGTM 👍

**Review by agnivade — APPROVED** [review]

**Review by metanerd — APPROVED** [review]
:+1:
